usaspending_api/references/management/commands/load_agencies.py

```diff
@@ -57,106 +63,144 @@ def handle(self, *args, **options):
-                            toptier_agency.cgac_code = frec_code
-                            toptier_agency.abbreviation = subtier_abbr
-                        else:
-                            toptier_agency.cgac_code = cgac_code
-                            toptier_agency.abbreviation = department_abbr
-
-                        toptier_agency.save()
-
-                    # Navy / Army / Air Force will just be a toptier, skip subtier
-                    if 'subsumed under dod' in subtier_code.lower():
-                        toptier_agency, created = ToptierAgency.objects.get_or_create(name=department_name)
-                        toptier_agency.cgac_code = cgac_code
-                        toptier_agency.abbreviation = department_abbr
-                        toptier_agency.save()
-
-                        agency, created = Agency.objects.get_or_create(toptier_agency=toptier_agency,
-                                                                       subtier_agency=None,
-                                                                       toptier_flag=True)
-                        agency.save()
-
-                        continue
-
-                    # Still need to grab the toptier for mapping
-                    if is_frec == 'TRUE':
-                        toptier_agency = ToptierAgency.objects.get(cgac_code=frec_code)
-                    else:
-                        toptier_agency = ToptierAgency.objects.get(cgac_code=cgac_code)
-
-                    # Sanity check
-                    assert(subtier_code.isalnum() and len(subtier_code) == 4)
-
-                    subtier_agency, created = SubtierAgency.objects.get_or_create(subtier_code=subtier_code)
-                    subtier_agency.name = subtier_name
-                    subtier_agency.abbreviation = subtier_abbr
-                    subtier_agency.save()
-
-                    agency, created = Agency.objects.get_or_create(toptier_agency=toptier_agency,
-                                                                   subtier_agency=subtier_agency,
-                                                                   toptier_flag=toptier_flag)
-                    agency.save()
-
-                with connection.cursor() as cursor:
-                    cursor.execute(MATVIEW_SQL)
-
-        except IOError:
-            self.logger.log("Could not open file to load from")
+            try:
+                self.unsaved_subtiers[toptier_code].append(subtier_dict)
+
+            except KeyError:
+                self.unsaved_subtiers[toptier_code] = [subtier_dict]
+
+            return None, None
+
+        subtier_agency, created = SubtierAgency.objects.get_or_create(subtier_code=subtier_code)
+        subtier_agency.name = subtier_name
+        subtier_agency.abbreviation = subtier_abbr
+        subtier_agency.save()
+
+        return toptier_agency, subtier_agency
+
+    def check_unsaved_subtiers(self, toptier_code):
```

Comment:
Just so I get the purpose of this function, is this is supposed to handle the cases for when a subtier row appears before the main toptier row appears?